src/media/local-audio-track.ts

```diff
@@ -0,0 +1,89 @@
+import { logger } from '../util/logger';
+import { LocalTrack } from './local-track';
+
+export type AudioDeviceConstraints = {
+  deviceId?: string;
+};
+
+export type StaticAudioEncoderConfig = {
+  low: '16kHz';
+};
+
+export type AudioEncoderConfig = {
+  bitrate?: number;
+  sampleRate?: number;
+  sampleSize?: number;
+  channelCount?: number;
+};
+
+export type MicrophoneConstraints = {
+  echoCancellation?: boolean;
+  autoGainControl?: boolean;
+  noiseSuppression?: boolean;
+  microphoneDeviceId?: string;
+  encoderConfig?: AudioEncoderConfig;
+};
```

Comment:
Are there built-in types these could leverage?

Reply:
we could but the naming is friendly and helps with documentation  , this is more cleaner for developer to understand they look similar to built ins

Reply:
done